src/User.js

```diff
@@ -1,19 +1,27 @@
-const Ingredient = require('../src/Ingredient.js');
+// const Ingredient = require('../src/Ingredient.js');
 
 class User {
   constructor(userData, ingredientsArray) {
     this.name = userData.name
     this.id = userData.id
-    this.pantry = userData.pantry.map(ingredient => new Ingredient(ingredient.ingredient, ingredient.amount, ingredientsArray))
-    this.favoriteRecipes = [];
-    this.recipesToCook = [];
+    this.pantry = userData.pantry.map(pantryItem => new Ingredient(pantryItem.ingredient, pantryItem.amount, ingredientsArray))
+    this.favoriteRecipes = JSON.parse(localStorage.getItem(`${this.id}-favorites`)) || [];
+    this.recipesToCook = JSON.parse(localStorage.getItem(`this${this.id}-recipes-to-cook}`)) || [];
   }
-  addRecipeToFavs() {
-
+  addRecipeToFavs(recipe) {
+    if (!this.favoriteRecipes.includes(recipe.id)) {
+      this.favoriteRecipes.push(recipe.id)
+      localStorage.setItem(`${this.id}-favorites`, JSON.stringify(this.favoriteRecipes))
+    }
```

Comment:
Nice I was working on this in the script files but it makes more sense here. Like how we are using local storage too